src/app/app.component.ts

```diff
@@ -1,10 +1,58 @@
-import { Component } from '@angular/core';
+import { Component, OnInit } from '@angular/core';
+
+class Point {
+  x: number; y: number;
+    constructor(x: number, y: number) {
+      this.x = x;
+      this.y = y;
+  }
+
+  add(newPoint: Point) {
+    return newPoint = new Point(newPoint.x + this.x, newPoint.y + this.y);
+  }
+}
+
+class Point3D {
```

Comment:
this should be a derived class from Point. Read the requirement again and try to make the changes.

Reply:
okay thank you, i just notice i had forgot to add it. I just upload the update.